modules/solid_mechanics/src/materials/IsotropicPowerLawHardening.C

```diff
@@ -0,0 +1,74 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "IsotropicPowerLawHardening.h"
+
+#include "SymmIsotropicElasticityTensor.h"
+
+#include "PiecewiseLinear.h"
+
+/**
+* Isotropic power law hardening material model. Before yield, the stress is youngs modulus * strain.
+* After yielding, the stress is K* pow(strain, n) where K is the strength coefficient,  n is the strain
+* hardening exponent and strain is the total strain.
+* Yield stress is the point of intersection of these two curves.
+**/
+
+template<>
+InputParameters validParams<IsotropicPowerLawHardening>()
+{
+  InputParameters params = validParams<IsotropicPlasticity>();
+
+  // Linear strain hardening parameters
+  params.set<Real>("yield_stress") = 1.0;
```

Comment:
The yield stress and the hardening function are calculated internally by this class. The isotropic plasticity class (from which Isotropic power law hardening is derived) requires the user to define the yield stress (constant or function) and hardening constant or function. I am setting the yield stress and hardening constant to 1.0 so that the base class does not complain.